src/agent.py

```diff
@@ -61,7 +61,10 @@ def _on_deprecated_agent_relation_joined(self, event: ops.RelationJoinedEvent) -
             event: The event fired from an agent joining the relationship.
         """
         container = self.charm.unit.get_container(JENKINS_SERVICE_NAME)
-        if not container.can_connect() or not self.state.is_storage_ready:
+        # This is to avoid the None type.
+        assert (binding := self.model.get_binding("juju-info"))  # nosec
```

Comment:
I am guessing the `get_binding` method will never return `None` here?
Can the comment be updated to include that? The current comment is a bit unclear.